File: Backend/src/user/access_control/definitions/modules/analytics.py
ANALYTICS_PERMISSIONS = {
    'analytics.manage': {
        'module': 'analytics',
        'action': 'manage',
        'display_name': 'Manage Website Analytics',
        'description': 'Full access to website and app visit analytics (page views, top pages, countries, traffic statistics)',
        'is_standalone': True,
    },
    'analytics.stats.manage': {
        'module': 'analytics',
        'action': 'manage',
        'display_name': 'Manage All Statistics',
        'description': 'Full access to all app statistics (users, admins, content, tickets, emails, system)',
    },
    'analytics.users.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View Users Statistics',
        'description': 'View detailed user statistics (sensitive)',
    },
    'analytics.admins.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View Admins Statistics',
        'description': 'View admin user statistics (highly sensitive)',
    },
    'analytics.content.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View Content Statistics',
        'description': 'View portfolio, blog, media statistics',
    },
    'analytics.tickets.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View Tickets Statistics',
        'description': 'View support tickets statistics and analytics',
    },
    'analytics.emails.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View Emails Statistics',
        'description': 'View email messages statistics and analytics',
    },
    'analytics.system.read': {
        'module': 'analytics',
        'action': 'read',
        'display_name': 'View System Statistics',
        'description': 'View system-level statistics (server, performance, logs)',
    },
}

File: admin-panel/src/core/permissions/config/routes/analytics.ts
import type { RouteRule } from "../types";

const createRule = (config: RouteRule): RouteRule => config;

export const analyticsRoutes: RouteRule[] = [
  createRule({
    id: "analytics-page-views",
    pattern: /^\/analytics\/?$/,
    module: 'analytics',
    action: "manage",
    description: "آمار بازدید وب‌سایت و اپلیکیشن",
  }),
  createRule({
    id: "analytics-stats",
    pattern: /^\/analytics\/stats\/?$/,
    module: 'analytics',
    action: "manage",
    description: "آمار سیستم و اپلیکیشن‌ها",
  }),
];
